library/L1_Peripheral/lpc40xx/capture.hpp

```diff
@@ -0,0 +1,285 @@
+
+        // Set Prescale register for Prescale Counter to milliseconds
+        uint32_t prescaler =
+            system_controller_
+                .GetPeripheralFrequency(timer_.channel.power_id) / frequency;
+        timer_.channel.timer_register->PR = prescaler;
+
+        // Enable timer
+        constexpr bit::Mask kTimerEnableBit = bit::CreateMaskFromRange(0);
+        timer_.channel.timer_register->TCR = bit::Set(
+                timer_.channel.timer_register->TCR, kTimerEnableBit.position);
+
+        // Install capture ISR
+        *timer_.channel.user_callback = isr;
+        interrupt_controller_.Register({
+                .interrupt_request_number = timer_.channel.irq,
+                .interrupt_service_routine = timer_.isr,
+                .enable_interrupt = true,
+                .priority = priority
+        });
+
+        // Disable both capture channels
+        ConfigureCapture(kCaptureChannel0, kCaptureEdgeModeNone);
+        ConfigureCapture(kCaptureChannel1, kCaptureEdgeModeNone);
+        EnableCaptureInterrupt(kCaptureChannel0, false);
+        EnableCaptureInterrupt(kCaptureChannel1, false);
+
+        return Status::kSuccess;
+    }
+
+    void GetCaptureStatus(CaptureChannelNumber channel, CaptureStatus_t &status)
+    {
+        status.flags = timer_.channel.timer_register->IR;
+
+        if (channel == kCaptureChannel0)
+        {
+          status.count = timer_.channel.timer_register->CR0;
+        }
+        else
+        {
+          status.count = timer_.channel.timer_register->CR1;
+        }
+    }
+
+    void ConfigureCapture(
+            CaptureChannelNumber channel,
+            CaptureEdgeMode mode) const override
+    {
+        if (channel == kCaptureChannel1)
+        {
+            timer_.channel.capture_pin1.SetPinFunction(3);
+        }
+        else
+        {
+            timer_.channel.capture_pin0.SetPinFunction(3);
+        }
+
+        static constexpr bit::Mask kModeBits[2] =
+        {
+            bit::CreateMaskFromRange(0, 1),
+            bit::CreateMaskFromRange(3, 4)
+        };
+
+        int which = 0;
+        if (channel == kCaptureChannel1)
+        {
+            which = 1;
+        }
+
+        timer_.channel.timer_register->CCR = bit::Insert(
+                timer_.channel.timer_register->CCR,
+                static_cast<uint32_t>(mode),
+                kModeBits[which]);
+    }
+
+    void EnableCaptureInterrupt(
+            CaptureChannelNumber channel, bool enabled) const override
+    {
+        static constexpr bit::Mask kPendingBits[2] =
+        {
+          bit::CreateMaskFromRange(4),
+          bit::CreateMaskFromRange(5)
+        };
+
+        static constexpr bit::Mask kEnableBits[2] =
+        {
+            bit::CreateMaskFromRange(2),
+            bit::CreateMaskFromRange(5)
+        };
+
+        int which = 0;
+        if (channel == kCaptureChannel1)
+        {
+            which = 1;
+        }
```

Comment:
If you set kCaptureChannel1 ==> 1 and kCaptureChannel0 == 0, then you can do this and omit the comparison.

```suggestion
        which = static_cast<int>(capture_channel_)
```